dashboard/test/controllers/script_levels_controller_test.rb

```diff
@@ -382,34 +382,50 @@ def script_level_params(script_level)
   end
 
   test "show: redirect to latest stable script version in family for logged out user if one exists" do
-    courseg_2018 = create :script, name: 'courseg-2018', family_name: 'courseg', version_year: '2018'
-    Script.stubs(:latest_stable_version).returns(courseg_2018)
+    courseg_2017 = create :script, name: 'courseg-2017', family_name: 'courseg', version_year: '2017', is_stable: true
+    create :script, name: 'courseg-2018', family_name: 'courseg', version_year: '2018', is_stable: true
+    create :script, name: 'courseg-2019', family_name: 'courseg', version_year: '2019'
 
-    courseg_2017 = create :script, name: 'courseg-2017', family_name: 'courseg', version_year: '2017'
     courseg_2017_stage_1 = create :stage, script: courseg_2017, name: 'Course G Stage 1', absolute_position: 1, relative_position: '1'
     courseg_2017_stage_1_script_level = create :script_level, script: courseg_2017, stage: courseg_2017_stage_1, position: 1
 
     get :show, params: {
-      script_id: courseg_2017.id,
+      script_id: courseg_2017.name,
       stage_position: courseg_2017_stage_1.relative_position,
       id: courseg_2017_stage_1_script_level.position,
     }
 
     assert_redirected_to '/s/courseg-2018?redirect_warning=true'
   end
 
+  test "show: does not redirect to latest stable script version if no_redirect query param provided" do
+    courseg_2017 = create :script, name: 'courseg-2017', family_name: 'courseg', version_year: '2017', is_stable: true
+    create :script, name: 'courseg-2018', family_name: 'courseg', version_year: '2018', is_stable: true
+
+    courseg_2017_stage_1 = create :stage, script: courseg_2017, name: 'Course G Stage 1', absolute_position: 1, relative_position: '1'
+    courseg_2017_stage_1_script_level = create :script_level, script: courseg_2017, stage: courseg_2017_stage_1, position: 1
+
+    get :show, params: {
+      script_id: courseg_2017.name,
+      stage_position: courseg_2017_stage_1.relative_position,
+      id: courseg_2017_stage_1_script_level.position,
+      no_redirect: "true"
+    }
+    assert_response :ok
```

Comment:
optional: it's a bit hard to see that this is an apples-to-apples comparison against the following test which does the same but without no_redirect. my recommendation would be to either extract common setup steps, or to just drop this get / assert at the end of the following test so that you can see the no_redirect param is making the difference.

this PR has dragged on a bit already, so feel free to skip and keep in mind for next time.